apiserver/facades/client/modelconfig/modelconfig.go

```diff
@@ -291,16 +294,8 @@ func (c *ModelConfigAPI) checkSecretBackend() state.ValidateConfigFunc {
 		if backendName == config.DefaultSecretBackend {
 			return nil
 		}
-		backend, err := c.backend.GetSecretBackend(backendName)
-		if err != nil {
-			return errors.Trace(err)
-		}
-		p, err := commonsecrets.GetProvider(backend.BackendType)
-		if err != nil {
-			return errors.Annotatef(err, "cannot get backend for provider type %q", backend.BackendType)
-		}
-		err = commonsecrets.PingBackend(p, backend.Config)
-		return errors.Annotatef(err, "cannot ping backend %q", backend.Name)
+		err := c.backendService.PingSecretBackend(ctx, backendName)
+		return errors.Annotatef(err, "cannot ping backend %q", backendName)
```

Comment:
There's a difference between the secret backend name not existing, and not being able to ping it. We should report those errors separately with different messages.